subprojects/packagefiles/libtomcrypt/0003.patch

```diff
@@ -0,0 +1,141 @@
+diff --git a/src/hashes/sha3.c b/src/hashes/sha3.c
+index c6faa0b..567d45e 100644
+--- a/src/hashes/sha3.c
++++ b/src/hashes/sha3.c
+@@ -174,6 +174,102 @@ int sha3_shake_init(hash_state *md, int num)
+    if (num != 128 && num != 256) return CRYPT_INVALID_ARG;
+    XMEMSET(&md->sha3, 0, sizeof(md->sha3));
+    md->sha3.capacity_words = (unsigned short)(2 * num / (8 * sizeof(ulong64)));
++   md->sha3.suffix = 0x1F;
++   return CRYPT_OK;
++}
++
++static unsigned long cshake_left_encode(unsigned long num, unsigned char *buf)
++{
++    unsigned int n, i;
++    size_t v;
++
++    for (v = num, n = 0; v && (n < sizeof(unsigned long)); n++, v >>= 8) {
```

Comment:
Why does `n` have to smaller than `sizeof(unsigned long)`? I am assuming this because you want to output less than a specified number of bytes to the output buffer (5?), and also it will always be true because `num` is either a small constant of the number of bits in a string. Or is this to make sure that `(num >> (8 * (n - i))` is not undefined behaviour if you shift more than the number of bits of `num`? In any case, I suggest to document that somewhere.

Reply:
this is just the reference implementation from https://github.com/XKCP/XKCP/blob/master/lib/high/Keccak/SP800-185/SP800-185.c